ebos/eclproblem.hh

```diff
@@ -1290,6 +1291,53 @@ public:
     }
 
 
+    std::unordered_map<std::string, double> fetchWellPI(int reportStep,
+                                                        const Opm::Action::ActionX& action,
+                                                        const Opm::Schedule& schedule,
+                                                        const std::vector<std::string>& matching_wells) {
+
+        auto wellpi_wells = action.wellpi_wells(WellMatcher(schedule[reportStep].well_order(),
+                                                            schedule[reportStep].wlist_manager()),
+                                                matching_wells);
+
+        if (wellpi_wells.empty())
+            return {};
+
+        const auto num_wells = schedule[reportStep].well_order().size();
+        std::vector<double> wellpi_vector(num_wells);
+        for (const auto& wname : wellpi_wells) {
+            if (this->wellModel_.hasWell(wname)) {
+                const auto& well = schedule.getWell( wname, reportStep );
+                wellpi_vector[well.seqIndex()] = this->wellModel_.wellPI(wname);
+            }
+        }
+
+        const auto& comm = this->simulator().vanguard().grid().comm();
+        if (comm.size() > 1) {
+            std::vector<double> wellpi_buffer(num_wells * comm.size());
+            comm.gather( wellpi_vector.data(), wellpi_buffer.data(), num_wells, 0 );
+            if (comm.rank() == 0) {
+                for (int rank=1; rank < comm.size(); rank++) {
+                    for (std::size_t well_index=0; well_index < num_wells; well_index++) {
+                        const auto global_index = rank*num_wells + well_index;
+                        const auto value = wellpi_buffer[global_index];
+                        if (value != 0)
+                            wellpi_vector[well_index] = value;
+                    }
+                }
+            }
+            comm.broadcast(wellpi_vector.data(), wellpi_vector.size(), 0);
```

Comment:
@blattms : Would you mind taking a look at this communication pattern please?  The objective is to communicate/distribute `num_wells` scalar values between all processes (ranks), with the additional provision that each process owns a subset of the values and that that subset may be partially replicated on other processes in the case of distributed wells.

Reply:
Communication seems correct, but seems a bit suboptimal (too much communication and too much space). Improvements should be made in a subsequent PR.

Seems like the goal here is that fetchWellPI returns a global vector (one entry for each well whether it is on the process or not) on each process. That is certainly safe, but we should ask whether that is needed (if actions on wells not on the process really influence the computation on the process.